pkg/auth/api/user/user_actions.go

```diff
@@ -170,3 +180,22 @@ func (h *Handler) refreshAttributes(actionName string, action *types.Action, req
 func (h *Handler) userCanRefresh(request *types.APIContext) bool {
 	return request.AccessControl.CanDo(v3.UserGroupVersionKind.Group, v3.UserResource.Name, "create", request, nil, request.Schema) == nil
 }
+
+func getMinimumPasswordLength() int {
+	return settings.PasswordMinLength.GetInt()
+}
+
+// validatePassword will ensure a password is the minimum required length, and that the password does not contain the username.
+func validatePassword(user string, pass string, minPassLen int) error {
+	passLen := len(pass)
+
+	if passLen < minPassLen {
+		return errors.Errorf("Password must be at least %v characters", minPassLen)
+	}
+
+	if user == pass {
+		return errors.New("Username and password are too similar")
```

Comment:
The message is rather vague as well. Suggest switching it to "Password cannot be the same as username".